### ISSUE TITLE
CORS missing allow origin on latest docker image

### ISSUE THREAD

**Issue by ditcheshurt** [presentator/presentator#182]
Pulled the latest docker image and ran through your docker install/setup guide and was presented with a red "Network Error" popup. Dev tools network tab showed a 400 error when making a call to /api/auth-methods.

This was done on a fresh VM install of Ubuntu Mate 18.04 desktop.

Any suggestions?

**Comment by ganigeorgiev** [comment]
The application itself doesn't have preconfigured CORS settings because most of the users doesn't need it.

To enable it, you could define the necessary headers in your nginx/apache config.
For example, if you are using a setup similar to the [docker-compose example](https://github.com/presentator/presentator-docker/tree/master/docker-compose-example) you could update the `Directory` section of your `vhosts.conf` like this:
```
# vhost.conf
...
<Directory /var/www/html/web>
    ...
    Header set Access-Control-Allow-Origin "*"
</Directory>
```

If you are still having troubles with the setup, feel free to let me know and I'll reopen the issue.

**Comment by ditcheshurt** [comment]
> The application itself doesn't have preconfigured CORS settings because most of the users doesn't need it.
> 
> To enable it, you could define the necessary headers in your nginx/apache config. For example, if you are using a setup similar to the [docker-compose example](https://github.com/presentator/presentator-docker/tree/master/docker-compose-example) you could update the `Directory` section of your `vhosts.conf` like this:
> 
> ```
> # vhost.conf
> ...
> <Directory /var/www/html/web>
>     ...
>     Header set Access-Control-Allow-Origin "*"
> </Directory>
> ```
> 
> If you are still having troubles with the setup, feel free to let me know and I'll reopen the issue.

So this was accessing it from localhost:8080 in firefow from the same vm desktop. Is there some configuration that needs to happen before? I fee like the defaults should at least allow a user that's installing this to access it from a localhost url (ie http://localhost:8080) without having to do anything additional after following the install guide.

I work as a developer at my day job so digging into the guts of isn't off putting but I think what you've got here is an amazing idea that has a lot of potential. I really liked testing it on the publicly available site but want to run an internal version of it.

I can see a huge set of additions that could be made to really make this software shine, I don't have a great deal of time to contribute to things but if I can get this running internally any modifications that are made I would love to share with the project. Especially since all the other versions out there that do basically the same thing are all pay-walled and ridiculously restrictive. Presentator would be a profound contribution to this software sector from the open source community for the people that just like self-hosting.

**Comment by ganigeorgiev** [comment]
Hm, could you elaborate more on your environment and installation steps that you've taken?

Just to be sure that we are on the same page (I admit that the multiple packages and their independent readmes maybe are a little confusing):

- For Development and local testing, the main readme's instructions should be followed - https://github.com/presentator/presentator#development. 

- For Production, the [presentator-starter](https://github.com/presentator/presentator-starter) template instructions should be followed. For convenience, the presentator-starter template has its own docker image - [presentator-docker](https://github.com/presentator/presentator-docker).

Internally, Presentator consists of 2 main "independent" composer packages - [SPA](https://github.com/presentator/presentator-spa) (aka. frontend) and [API](https://github.com/presentator/presentator-api) (aka. backend).
Since presentator-starter (and its related docker image) wraps both packages in a single php application no CORS should be necessary out of the box, because they both will be accessible from a single host address (eg. example.com serves the SPA and example.com/api - the API).

You need to configure CORS only if you want 3rd party websites/services or plugins (eg. the [Figma plugin](https://github.com/presentator/presentator-figma)) to be able to make request to the api.

**Comment by ganigeorgiev** [comment]
_Actually, I stand corrected. I forgot about this because I use nginx most of the times, but CORS is enabled by default for apache environments by the created default `.htaccess` during the application installation - https://github.com/presentator/presentator/blob/master/packages/api/environments/dev/web/.htaccess_

**Comment by ganigeorgiev** [comment]
_Closing due to lack of activity._

@ditcheshurt if you are still facing issues with the setup, feel free to comment or open a new issue.
